pkg/delivery/fcm.go

```diff
@@ -0,0 +1,84 @@
+package delivery
+
+import (
+	"context"
+	"log"
+
+	firebase "firebase.google.com/go/v4"
+	"firebase.google.com/go/v4/messaging"
+	"github.com/xmtp/example-notification-server-go/pkg/options"
+	"go.uber.org/zap"
+	"google.golang.org/api/option"
+)
+
+type FcmDelivery struct {
+	logger *zap.Logger
+	client *messaging.Client
+}
+
+func NewFcmDelivery(ctx context.Context, logger *zap.Logger, opts options.FcmOptions) (*FcmDeliveryAgent, error) {
+	creds := option.WithCredentialsJSON([]byte(opts.CredentialsJson))
+	app, err := firebase.NewApp(ctx, &firebase.Config{
+		ProjectID: opts.ProjectId,
+	}, creds)
+	if err != nil {
+		log.Fatalf("error initializing app: %v\n", err)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	messaging, err := app.Messaging(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &FcmDelivery{
+		logger: logger,
+		client: messaging,
+	}, nil
+}
+
+func (f *FcmDelivery) Send(ctx context.Context, token, topic, message string) error {
+	title := "New message from XMTP"
+	body := "Open app to read"
+	data := map[string]string{
+		"topic":            topic,
+		"encryptedMessage": message,
+	}
+
+	_, err := f.client.Send(ctx, &messaging.Message{
```

Comment:
I tried just throwing the kitchen sink at this with all possible arguments. Will test with a real FCM account and see if it actually allows me to send this whole payload if the FCM account is only partially setup (for example, configuring APNS options here without having APNS setup in Firebase dashboard). If that doesn't work, will have to create a more involved setup where you can configure what services are included in the payload or not.